src/UserSecretKey.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace MultiversX;
+
+use Exception;
+use Elliptic\EdDSA;
+use Illuminate\Support\Collection;
+
+class UserSecretKey
+{
+    const USER_SEED_LENGTH = 32;
+    const USER_PUBKEY_LENGTH = 32;
+
+    private function __construct(
+        public readonly string $key,
+    ) {
+    }
+
+    public static function fromPem(string $text, int $index = 0): UserSecretKey
+    {
+        return static::parseUserKeysFromPem($text)[$index];
+    }
+
+    public function sign(string $message): string
+    {
+        return (new EdDSA('ed25519'))
+            ->keyFromSecret($this->key)
+            ->sign($message)
+            ->toHex();
+    }
+
+    private static function parseUserKeysFromPem(string $text): Collection
+    {
+        return static::parsePem($text, static::USER_SEED_LENGTH + static::USER_PUBKEY_LENGTH)
+            ->map(fn (string $buffer) => new UserSecretKey(bin2hex(substr($buffer, 0, static::USER_SEED_LENGTH))));
+    }
+
+    private static function parsePem(string $text, int $expectedLength): Collection
+    {
+        // converted from https://github.com/multiversx/mx-sdk-js-wallet/blob/main/src/pem.ts
```

Comment:
Maybe also include a release tag in the link (since this might change upon alignment to specs).

Reply:
Done